### DISCUSSION TITLE
Python 3.12 type aliases are not processed correctly

### DISCUSSION THREAD

**Discussion by palle-k** [fastapi/fastapi#13083] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import FastAPI, Body
from pydantic import BaseModel

class Foo(BaseModel):
    bar: str

type FooType = Foo

app = FastAPI()

@app.post("/a")
async def foo(body: FooType):
    return body

@app.post("/b")
async def foo(body: Foo):
    return body

@app.post("/c")
async def foo(body: Annotated[FooType, Body()]):
    return body
```

### Description

When declaring a method, where the body parameter is an instance of TypeAliasType, FastAPI incorrectly assumes that the parameter is a query parameter, not a body parameter.

- /a: FastAPI (incorrectly) expects query parameter
- /b: FastAPI (correctly) expects body parameter
- /c: Workaround: FastAPI (correctly) expects body parameter

### Operating System

macOS

### Operating System Details

macOS 15.2 (24C101)

### FastAPI Version

0.115.6

### Pydantic Version

2.10.3

### Python Version

Python 3.12.8

**Comment by Kludex** [comment, marked as answer]
- https://github.com/fastapi/fastapi/issues/10719
